### PR TITLE
Fix false negatives for grid shorthands in named-grid-areas-no-invalid

### PR DESCRIPTION
**Pull request by jeddy3**
> Which issue, if any, is this issue related to?

Ref: https://github.com/stylelint/stylelint/pull/5513#pullrequestreview-746590742

> Is there anything in the PR that needs further explanation?

[`grid`](https://developer.mozilla.org/en-US/docs/Web/CSS/grid) and [`grid-template`](https://developer.mozilla.org/en-US/docs/Web/CSS/grid-template).

### CODE REVIEW

File: lib/rules/named-grid-areas-no-invalid/index.js [hunk 1]
```diff
@@ -25,7 +25,7 @@ const rule = (primary) => {
 			return;
 		}
 
-		root.walkDecls(/^grid-template-areas$/i, (decl) => {
+		root.walkDecls(/^grid$|^grid-template|$^grid-template-areas|$/i, (decl) => {
```
Comment (ybiquitous) [inline lib/rules/named-grid-areas-no-invalid/index.js:28]: [suggestion] Can we simplify this regex as below?

```suggestion
		root.walkDecls(/^(?:grid|grid-template|grid-template-areas)$/i, (decl) => {
```

File: lib/rules/named-grid-areas-no-invalid/__tests__/index.js [hunk 2]
```diff
@@ -103,8 +112,8 @@ testRule({
 		},
 		{
 			code: stripIndent`
-				a { grid-template-areas: 
-			   	""; 
+				a { grid-template-areas:
+			   	"";
```
Comment (ybiquitous) [inline lib/rules/named-grid-areas-no-invalid/__tests__/index.js:116]: [suggestion] The whitespaces for indentation seem to be wrong.

```suggestion
					"";
```

**Review by ybiquitous — COMMENTED** [review]
@jeddy3 Thank you for creating the PR. I've left some refactoring suggestions, but it looks almost fine! 😄

File: lib/rules/named-grid-areas-no-invalid/index.js (hunk 1 above)
Comment (jeddy3) [reply lib/rules/named-grid-areas-no-invalid/index.js:28]: I had a feeling it could be better written as I was writing it :)

**Comment by jeddy3** [comment]
Thanks for the review and suggestions. I've committed them both.

**Comment by ybiquitous** [comment]
My suggestion code seems to be something wrong. 👀

**Comment by ybiquitous** [comment]
By the change of indentation (80f5ab3), it is necessary to change the expectation of `column`:

```diff
diff --git a/lib/rules/named-grid-areas-no-invalid/__tests__/index.js b/lib/rules/named-grid-areas-no-invalid/__tests__/index.js
index e49bc06d..4855c7fd 100644
--- a/lib/rules/named-grid-areas-no-invalid/__tests__/index.js
+++ b/lib/rules/named-grid-areas-no-invalid/__tests__/index.js
@@ -117,7 +117,7 @@ testRule({
 				}`,
 			message: messages.expectedToken(),
 			line: 2,
-			column: 4,
+			column: 2,
 		},
 		{
 			code: 'a { grid-template-areas: "" "" ""; }',
```

**Review by ybiquitous — APPROVED** [review]
@jeddy3 Sorry for the inconvenience. LGTM! 👍🏼

**Comment by jeddy3** [comment]
No worries at all! I appreciate the review 😄

**Comment by jeddy3** [comment]
> - Fixed: `named-grid-areas-no-invalid` false negatives for `grid` and `grid-template` shorthand properties ([#5514](https://github.com/stylelint/stylelint/pull/5514)).

**Pull request merged**